python/mlx/nn/layers/linear.py

```diff
@@ -26,20 +41,92 @@ class Linear(Module):
 
     def __init__(self, input_dims: int, output_dims: int, bias: bool = True):
         super().__init__()
-        scale = math.sqrt(1 / input_dims)
+        self.input_dims = input_dims
+        self.output_dims = output_dims
+        self.weight = mx.zeros((output_dims, input_dims))
+        if bias:
+            self.bias = mx.zeros((output_dims,))
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        scale = math.sqrt(1.0 / self.input_dims)
         self.weight = mx.random.uniform(
             low=-scale,
             high=scale,
-            shape=(output_dims, input_dims),
+            shape=(self.output_dims, self.input_dims),
         )
+        if "bias" in self:
+            self.bias = mx.random.uniform(
+                low=-scale,
+                high=scale,
+                shape=(self.output_dims,),
+            )
+
+    def _extra_repr(self):
+        return f"input_dims={self.input_dims}, output_dims={self.output_dims}, bias={'bias' in self}"
+
+    def __call__(self, input):
+        output = input @ self.weight.T
+        if "bias" in self:
+            output = output + self.bias
+        return output
+
+
+class Bilinear(Module):
+    r"""Applies a bilinear transformation to the input.
+
+    Concretely:
+
+    .. math::
+
+        y = input1^\top W input2 + b
```

Comment:
x_1 and x_2 in the math will read more cleanly.

Reply:
I've changed the math to x_1 and x_2 https://github.com/ml-explore/mlx/pull/315/commits/a4b43db9af5a00285213d7c9ae710166663aca58 (Line 80)